### ISSUE TITLE
props-name-convention: eslint-plugin-awscdk only checks classes that directly extend Construct (diverges from oxlint-plugin-awscdk)

### ISSUE THREAD

**Issue by ren-yamanashi** [ren-yamanashi/eslint-cdk-plugin#604]
### Describe the bug

In eslint-plugin-awscdk, `props-name-convention` only fires when the class literally has `extends Construct`. A Construct that extends a project base class (`class Derived extends BaseConstruct`, where `BaseConstruct extends Construct`) is never checked. oxlint-plugin-awscdk walks the inheritance chain and reports it.

`src/props-name.ts`:

```ts
import { Construct } from "constructs";

export class BaseConstruct extends Construct {
  constructor(scope: Construct, id: string) {
    super(scope, id);
  }
}

export interface Props {
  readonly value: string;
}

export class Derived extends BaseConstruct {
  constructor(scope: Construct, id: string, props: Props) {
    super(scope, id);
    console.log(props.value);
  }
}

export class Direct extends Construct {
  constructor(scope: Construct, id: string, props: Props) {
    super(scope, id);
    console.log(props.value);
  }
}
```

Rule enabled: `awscdk/props-name-convention: "error"`.

```
$ npx eslint src/props-name.ts
  21:52  error  Props interface name 'Props' should follow '${ConstructName}Props' format. Expected 'DirectProps'  awscdk/props-name-convention

$ npx oxlint --format=unix -c .oxlintrc.json src/props-name.ts
src/props-name.ts:14:52: Props interface name 'Props' should follow '${ConstructName}Props' format. Expected 'DerivedProps'. [Error/awscdk(props-name-convention)]
src/props-name.ts:21:52: Props interface name 'Props' should follow '${ConstructName}Props' format. Expected 'DirectProps'. [Error/awscdk(props-name-convention)]
```

### Expected Behavior

Both plugins report lines 14 and 21: `Derived` extends `Construct` through `BaseConstruct`, and every other class-level rule already treats it as a Construct.

### Current Behavior

eslint-plugin-awscdk reports only line 21 (the direct `extends Construct`).

### Plugin

eslint-plugin-awscdk

### Plugin Version

eslint-plugin-awscdk 4.4.2 (compared against oxlint-plugin-awscdk 1.0.2)

### Language Version

TypeScript 6.0.3

### Additional Context

```yaml
# machine-readable summary
rules: ["props-name-convention"]
symptom: ["false-negative", "cross-plugin-divergence"]
affected_packages: ["eslint-plugin-awscdk"]
root_cause: "packages/eslint/src/rules/props-name-convention.ts:33-34 (the type of `node.superClass` is the constructor type `typeof BaseConstruct`; `Type.getBaseTypes()` returns undefined for it, so only a superClass whose symbol is literally `Construct` matches)"
verified_on: "main @ 9caea55 (eslint-plugin-awscdk 4.4.2, oxlint-plugin-awscdk 1.0.2)"
related_issues: ["#536", "#570"]
```

The other class-level rules resolve `getTypeAtLocation(node)` (the instance type of the class declaration) and walk its base types correctly; this rule is the only one resolving the `superClass` expression. Verified fix direction: resolving `getTypeAtLocation(node)` (the class instance type) instead of `node.superClass` makes the eslint plugin report both lines, with `examples/` still at 43 errors. The divergence is mentioned as known in `.claude/decisions/2026-09-02-two-tier-ignore-list.md` but was not tracked anywhere; this issue tracks it.

Reproduction environment:

- macOS 14.1, Node.js v24.18.0, pnpm 10.29.3
- aws-cdk-lib 2.268.0 / constructs 10.8.1
- ESLint 10.9.1 + typescript-eslint 8.69.0 + TypeScript 6.0.3
- oxlint 1.81.0 + oxlint-tsgolint 7.0.2001 (corsa-oxlint 1.13.1)
